ftw/simplelayout/images/validators.py

```diff
@@ -0,0 +1,95 @@
+from ftw.simplelayout import _
+from ftw.simplelayout.images.interfaces import IImageLimits
+from plone import api
+from z3c.form import validator
+from zope.interface import Invalid
+
+
+class ImageLimitValidator(validator.SimpleFieldValidator):
+    """Validates the image-dimensions.
+
+    Soft-limit: Validates if the given image has at least the dimensions given
+    within the ImageLimit-configuration for the current contenttype.
+
+    Hard-limit: Validates if the given image has at least the dimensions given
+    within the ImageLimit-configuration for the current contenttype.
+    """
+    identifier = None
+
+    def __init__(self, *args, **kwargs):
+        super(ImageLimitValidator, self).__init__(*args, **kwargs)
+        self.image_limits = IImageLimits(self.context)
+
+    def validate(self, value):
+        super(ImageLimitValidator, self).validate(value)
+        if not value:
+            return
+
+        self._validate_hard_limit(value)
+        self._validate_soft_limit(value)
+
+    def _validate_hard_limit(self, value):
+        if not self._validate_limit_for('hard', value):
+            raise Invalid(self._image_hard_limit_invalid_error_message(value))
+
+    def _validate_soft_limit(self, value):
+        if not self._validate_limit_for('soft', value):
+            api.portal.show_message(
```

Comment:
We also have the static message on the image of the textblock. When does this limit occur? Dies it conflict with the message in the macro?

Reply:
The soft-limit portal-message will be visible for contenttypes not loaded within an overlay.

Currently, this is not the case. The message will not be shown anywhere. The Block-Implementation discards the portalmessages after creating the block. But I saw, that editing the block will not discard it. So I have to fix this anyway.

Currently, the `lowImageQualityIndicator`macro only shows the text "low image quality". But it would be helpful, if the user knows, what dimensions his image should be to match the requirements. So I could move the warning-message to the `lowImageQualityIndicator`:

Before:
<img width="311" alt="bildschirmfoto 2018-08-23 um 09 44 42" src="https://user-images.githubusercontent.com/557005/44511900-31d4f280-a6b9-11e8-8834-e7e790865a90.png">

After:
<img width="324" alt="bildschirmfoto 2018-08-23 um 09 41 09" src="https://user-images.githubusercontent.com/557005/44511908-37323d00-a6b9-11e8-958d-596d278f18a8.png">

Another approach is to show the required dimensions within a tooltip after hovering the `lowImageQualityIndicator`

What is your opinion?